modules/generator/http.go

```diff
@@ -0,0 +1,42 @@
+package generator
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/gogo/protobuf/jsonpb"
+	"github.com/grafana/tempo/pkg/api"
+	"github.com/grafana/tempo/pkg/tempopb"
+	"github.com/opentracing/opentracing-go"
+)
+
+func (g *Generator) SpanSummaryHandler(w http.ResponseWriter, r *http.Request) {
+	ctx, cancel := context.WithDeadline(r.Context(), time.Now().Add(g.cfg.SummaryTimeout))
```

Comment:
Perhaps we just handle the context here and avoid the generator making a timeout.

Reply:
r.context ?